### PR TITLE
Bug 1318474 - Store job metadata in main database

### PR DESCRIPTION
**Pull request by wlach**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1985)

### CODE REVIEW

File: treeherder/webapp/api/jobs.py [hunk 1]
```diff
@@ -93,56 +180,93 @@ def list(self, request, project, jm):
...
                 try:
-                    # ensure last_modified is a date
                     parser.parse(datestr)
                 except ValueError:
                     return Response(
                         "Invalid date value for `last_modified`: {}".format(datestr),
                         status=HTTP_400_BAD_REQUEST)
 
+        offset = int(filter_params.get("offset", 0))
+        count = int(filter_params.get("value", 10))
```
Comment (camd) [inline treeherder/webapp/api/jobs.py]: shouldn't this be ``filter_params.get("count", 10)`` rather than ``"value"`` ?

File: treeherder/webapp/api/jobs.py [hunk 2]
```diff
@@ -327,30 +451,51 @@ def bug_suggestions(self, request, project, pk=None):
...
         # let's cap it to 50 elements
-        count = min(int(url_query_filter.pop("count", 10)), 50)
-        return_type = url_query_filter.pop("return_type", "dict").lower()
-        results = jm.get_job_list_sorted(offset, count,
-                                         conditions=url_query_filter.conditions)
-
-        # add result set id for backwards compatibility
-        for result in results:
-            result['result_set_id'] = result['push_id']
+        count = int(filter_params.get("value", 50))
```
Comment (camd) [inline treeherder/webapp/api/jobs.py]: Hmm, here's another example of ``value`` being used.  Is this still going to support the ``count`` param being passed in the querystring?  I know it's used by people.  Sorry if I'm just ignorant of this feature, but I can't find an example of ``value`` meaning ``count`` in the Yahoo!s / DRF docs.

I see ``limit`` here: http://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination

Comment (wlach) [inline treeherder/webapp/api/jobs.py]: I think this was a typo. Bad that the unit tests didn't catch it, I'll add one before committing.

File: treeherder/model/derived/jobs.py [hunk 3]
```diff
@@ -656,6 +672,9 @@ def store_job_data(self, data):
                 debug_show=self.DEBUG,
                 placeholders=coalesced_job_guid_placeholders,
                 executemany=True)
+            for (coalesced_to_guid, job_guid) in coalesced_job_guid_placeholders:
```
Comment (wlach) [inline treeherder/model/derived/jobs.py]: This doesn't actually instantiate a new job object, the filter/update gets reduced down to an sql update statement after Django has had its way with it. I'm not particularly worried about this code being inefficient vs. everything else in this jobs model. My feeling is that we should leave it as-is, and let someone else do the profiling of this code if/when it becomes a problem. If history is anything to go by, the bottlenecks will not be where we think they are.

File: treeherder/model/derived/jobs.py [hunk 4]
```diff
@@ -943,6 +962,26 @@ def _load_job(self, job_datum, push_id, lower_tier_signatures):
...
             project_specific_id=ds_job_id,
             defaults={
+                'signature': signature,
+                'build_platform': build_platform,
+                'machine_platform': machine_platform,
+                'machine': machine,
+                'option_collection_hash': option_collection_hash,
+                'job_type': job_type,
+                'product': product,
+                'failure_classification_id': 1,
```
Comment (wlach) [inline treeherder/model/derived/jobs.py]: Yeah probably doesn't hurt to change / fix this. :)

Comment (wlach) [inline treeherder/model/derived/jobs.py]: Fixed (landed in another PR)

File: treeherder/model/derived/jobs.py [hunk 5]
```diff
@@ -298,17 +287,17 @@ def trigger_all_talos_jobs(self, requester, push_id, times):
...
         )
 
-    def _job_action_event(self, job, action, requester):
+    def _job_action_event(self, job_id, action, requester):
         """
         Helper for issuing an 'action' for a given job (such as
         cancel/retrigger)
 
-        :param job dict: The job which this action was issued to.
+        :param job int: The id of job which this action was issued to.
```
Comment (wlach) [inline treeherder/model/derived/jobs.py]: Fixed

File: treeherder/webapp/api/jobs.py (hunk 1 above)
Comment (wlach) [inline treeherder/webapp/api/jobs.py]: Fixed (and added unit tests)

**Comment by wlach** [comment]
This PR has been applied in pieces, so we can close this.
